### ISSUE TITLE
xtest crashed at regression_1008.4 Load corrupt TA in qemu

### ISSUE THREAD

**Issue by SarahL21** [OP-TEE/optee_os#2506]
Hello, 
I have built an RPMB secure storage TA but xtest crashes.. I have enabled CFG_RPMB_FS and called the TEE internal API functions with TEE_STORAGE_PRIVATE_RPMB.. I have checked regression_1000.c for more info about the error message, I found it's something related to CFG_SECSTOR_TA_MGMT_PTA: as I know SECSTOR is for anti rollback protection.. 
Has anyone encountered such an issue before? 
Any help or tip would be appreciated, 
Thanks

**Comment by jbech-linaro** [comment]
> E/TC:? 0 tee_ta_open_session:586 Failed. Return error 0xffff0000

That line corresponds to 
```c
#define TEE_ERROR_GENERIC                 0xFFFF0000
```
However, in I cannot see that error code being set anywhere in the function [tee_ta_open_session](https://github.com/OP-TEE/optee_os/blob/master/core/kernel/tee_ta_manager.c#L524-L589)? But on the normal path when busy, the call to [enter_open_session](https://github.com/OP-TEE/optee_os/blob/master/core/kernel/tee_ta_manager.c#L562) will update `res`. But if this is wrong, then it will automatically close the session [here](https://github.com/OP-TEE/optee_os/blob/master/core/kernel/tee_ta_manager.c#L573-L574). Which is probably why we see the line below in the log you pasted:

```bash
D/TC:? 0 tee_ta_close_session:380 tee_ta_close_session(0xe187dd8)
```

After that everything just bails out. I think you should debug your 
```c
host:  TEEC_OpenSession(...)
TA: TA_OpenSessionEntryPoint(...)
```
code path, because that is where I suspect that things are going wrong.

**Comment by jbech-linaro** [comment]
@SarahL21 issue fixed in #2525 .
